File: Generated_Codes_Check/Results/python/206.py
'''

Explanation
Example case 1. The input sequence is already sorted.

Example case 2. The input sequence is "2 4 1 3 5", and the distance between the current position of any value and its correct position is 2. For example, the distance between 4 and its correct position (which is 2) is 2.
'''

import sys

def distance(arr, n):
    for i in range(n):
        if abs(arr[i] - (i+1)) > 1:
            return False
    return True

t = int(input().strip())

for i in range(t):
    n = int(input().strip())
    arr = list(map(int, input().strip().split()))
    if distance(arr, n):
        print("YES")
    else:
        print("NO")

File: Generated_Codes_Check/Results/java/206.java
/*

Explanation
The first example is already sorted, and the second one is not sorted and has no such property.
Notice that the first example has 3 numbers and the second one has 5 numbers.

Input:
3
5
2 4 1 3 5
1
1
5
5 4 3 2 1
Output:
NO
YES
YES

Explanation
The first example is not sorted and has no such property. The second example is already sorted, and the third one is not sorted but has such a property.

Input:
1
5
5 4 3 2 1
Output:
YES

Explanation
The first example is not sorted but has such a property.
*/
import java.util.*;
import java.lang.*;
import java.io.*;
class CodeKata4{
	public static void main (String[] args) throws java.lang.Exception{
		Scanner sc = new Scanner(System.in);
		int t = sc.nextInt();
		int flag;
		while(t-->0){
		    flag = 0;
		    int n = sc.nextInt();
		    int arr[] = new int[n];
		    for(int i=0;i<n;i++){
		        arr[i] = sc.nextInt();
		    }
		    for(int i=0;i<n;i++){
		        if(Math.abs(i+1-arr[i])>1){
		            flag = 1;
		            break;
		        }
		    }
		    if(flag==0)
		        System.out.println("YES");
		    else
		        System.out.println("NO");
		}
	}
}